app/src/main/java/com/mailerdaemon/app/LoginActivity.kt

```diff
@@ -43,31 +39,26 @@ class LoginActivity: AppCompatActivity() {
         if (getSharedPreferences("MAIN", Context.MODE_PRIVATE).getBoolean("intro", true)) {
             startActivity(Intent(this, IntroActivity::class.java))
             finish()
-        } else {
-            val currentUser = mAuth.currentUser
-            currentUser?.let { startMain() }
-        }
+        } else
+            startMain()
         setContentView(R.layout.activity_login)
         progress_bar.visibility = View.GONE
         forgot_password.setOnClickListener { startActivity(Intent(this, ForgotPassActivity::class.java)) }
         login.setOnClickListener {
-            val email = Objects.requireNonNull(login_email.text).toString().trim { it <= ' ' }
-            val password = Objects.requireNonNull(login_password.text).toString()
+            val email = login_email!!.text.toString().trim { it <= ' ' }
+            val password = (login_password!!.text).toString()
             if (email.isNotEmpty()) {
                 if (password.isNotEmpty()) {
                     progress_bar.visibility = View.VISIBLE
                     mAuth.signInWithEmailAndPassword(email, password).addOnCompleteListener { task: Task<AuthResult?> ->
                         progress_bar.visibility = View.GONE
-                        if (task.isSuccessful) {
-                            saveUser(Objects.requireNonNull(mAuth.currentUser))
-                        } else {
-                            Toast.makeText(applicationContext, "Invalid password", Toast.LENGTH_LONG).show()
-                        }
+                        if (task.isSuccessful)
+                            saveUser((mAuth.currentUser)!!)
+                         else
+                            toast("Invalid password")
```

Comment:
Not this toast. I don't know how this library is still present but don't use `anko.toast`.  Use context.toast extention function which is present in Utils.kt file